ui/raidboss/data/04-sb/trial/lakshmi.js

```diff
@@ -0,0 +1,338 @@
+      outputStrings: {
+        text: {
+          en: 'USE VRIL OR DIE',
+        },
+      },
+    },
+    {
+      id: 'Lakshmi Divine Denial',
+      netRegex: NetRegexes.startsUsing({ id: '2485', source: 'Lakshmi', capture: false }),
+      netRegexDe: NetRegexes.startsUsing({ id: '2485', source: 'Lakshmi', capture: false }),
+      netRegexFr: NetRegexes.startsUsing({ id: '2485', source: 'Lakshmi', capture: false }),
+      netRegexJa: NetRegexes.startsUsing({ id: '2485', source: 'ラクシュミ', capture: false }),
+      netRegexCn: NetRegexes.startsUsing({ id: '2485', source: '吉祥天女', capture: false }),
+      netRegexKo: NetRegexes.startsUsing({ id: '2485', source: '락슈미', capture: false }),
+      alertText: (data, _, output) => output.text(),
+      outputStrings: {
+        text: {
+          en: 'Vril + Knockback',
+          de: 'Vril + Rückstoß',
+          fr: 'Vril + Poussée',
+          ja: 'エーテル + 完全なる拒絶',
+          cn: '完全拒绝',
+          ko: '락슈미 에테르 + 넉백',
+        },
+      },
+    },
+    {
+      // Nobody with a different marker should be told to stack.
+      id: 'Lakshmi Headmarker Collect',
+      netRegex: NetRegexes.headMarker({ }),
+      run: (data, matches) => {
+        data.avoidStack = data.avoidStack || [];
+        if (matches.id !== '003E')
+          data.avoidStack.push(matches.target);
+      },
+    },
+    {
+      // Activating on any use of Hand of Beauty/Grace or Alluring Arm.
+      // Head markers don't appear until the end of the castbar,
+      // and every head marker section begins with one of these abilities,
+      // so this should be perfectly safe.
+      id: 'Lakshmi Headmarker Cleanup',
+      netRegex: NetRegexes.startsUsing({ id: ['2486', '2487', '2488'], source: 'Lakshmi', capture: false }),
+      netRegexDe: NetRegexes.startsUsing({ id: ['2486', '2487', '2488'], source: 'Lakshmi', capture: false }),
+      netRegexFr: NetRegexes.startsUsing({ id: ['2486', '2487', '2488'], source: 'Lakshmi', capture: false }),
+      netRegexJa: NetRegexes.startsUsing({ id: ['2486', '2487', '2488'], source: 'ラクシュミ', capture: false }),
+      netRegexCn: NetRegexes.startsUsing({ id: ['2486', '2487', '2488'], source: '吉祥天女', capture: false }),
+      netRegexKo: NetRegexes.startsUsing({ id: ['2486', '2487', '2488'], source: '락슈미', capture: false }),
+      run: (data) => delete data.avoidStack,
+    },
+    { // Stack marker
+      id: 'Lakshmi Pall of Light',
+      netRegex: NetRegexes.headMarker({ id: '003E' }),
+      delaySeconds: 0.5,
+      alertText: function(data, matches, output) {
+        if (!data.avoidStack.includes(data.me))
+          return;
+        return output.dontStack();
+      },
+      infoText: function(data, matches, output) {
+        if (data.avoidStack.includes(data.me))
+          return;
+        if (data.me === matches.target)
+          return output.stackOnYou();
+        return output.stack({ target: matches.target });
+      },
+      outputStrings: {
+        dontStack: {
```

Comment:
I don't know if I fully remember the fight, but I'm assuming based on the logic here that you will have two triggers at the same time, one telling you Cross Marker and the other saying not to stack? Does that feel like too much?

Reply:
It sort of is, but the issue is that at least the circle markers have environmental effects when they resolve, and it's *really* easy for people to autopilot based on triggers. (I dislike it when people run from stack markers, but I get *really* frustrated when people with spread markers attempt to stack.)

The way I see it, each of the non-stack markers has a specific thing they need to *do*, and they should do that and not even attempt to stack. Adding "don't stack" doesn't always make sense since the stack marker isn't always up when the circle/cross/cone marker are, so we would then need to special-case each one based on whether or not there's a stack marker on the field. This then *further* means we need to do a collect/call setup for each marker, rather than just the stack marker.

Conversely, every time there's a stack marker, it's accompanied by *something* else, so it makes sense to warn those other people off from within the stack trigger, at least to me.